event_model/__init__.py

```diff
@@ -2000,31 +2000,24 @@ def compose_resource(
 @dataclass
 class ComposeStreamDatum:
     stream_resource: StreamResource
-    stream_name: str
     counter: Iterator
 
     def __call__(
         self,
-        datum_kwargs: Dict[str, Any],
-        event_count: int = 1,
-        event_offset: int = 0,
```

Comment:
Let's put `descriptor: Optional[EventDescriptor]` it in here, but default it to `None` so it can be created in the Ophyd Device without it, and the descriptor uid added later by bluesky

Reply:
You think it should be in the `__call__` kwargs?  You mean adding an optional `descriptor_uid` string field to the `StreamDatum`?